src/olympia/reviewers/utils.py

```diff
@@ -1009,15 +1012,25 @@ def create_decision():
         else:
             create_decision()
 
-        self.log_action(
-            activity_action,
-            decisions=decisions,
-            reasons=reasons,
-            policies=policies,
-            **(log_action_kw or {}),
-        )
-        self.update_queue_history()
+        log_entry = None
+        if action_completed:
+            log_entry = self.log_action(
+                activity_action,
+                decisions=decisions,
+                reasons=reasons,
+                policies=policies,
+                **(log_action_kw or {}),
+            )
+            self.update_queue_history(log_entry)
         for decision in decisions:
+            log_entry = decision.execute_action()
```

Comment:
This is confusing me at first glance. Shouldn't it be inside the `if not action_completed` condition below ?

Reply:
`execute_action` also clears or sets pending rejection fks with the cinder job, plus clears cinder related NHRs.  That logic _could_ be moved into this class instead (reversing the trend)

Reply:
Got it. I think it's fine to have that in `execute_action()`, it's just that while we're in this weird in-between state with stuff being performed on both sides depending on the action it's kinda confusing which part does what